### DISCUSSION TITLE
MSSQL - ReadOnly Replicas

### DISCUSSION THREAD

**Discussion by mattyoder77** [prisma/prisma#23604] [Q&A]
### Question

how does prisma create a connection using tiberius? I'm connecting to an availability group that has read only routing setup and I can connect to it using SSMS but applying the same connection string to prisma is connecting to the primary instead of the secondary replica. Looking at the tiberius repo (specifically https://github.com/prisma/tiberius/pull/297) it does seem to have this functionality so I'm wondering if prisma is doing something.

### How to reproduce (optional)

Using connection string 
```
DATABASE_URL="sqlserver://MY_IP:1433;user=MY_USER;password=MY_PASS;TrustServerCertificate=true;database=MY_DB;ApplicationIntent=ReadOnly"
```

### Expected behavior (optional)

When it connects to the availability group on sql server it should use the read only replica when the connection string contains ApplicationIntent=ReadOnly

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @myoder020 👋 

Thank you for raising this question.

I do not know the implementation details for how Prisma handles connection with Tiberius. There is an [extension](https://github.com/prisma/extension-read-replicas) you could use for working with Read replicas. You can learn more about it in the [documentation](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/read-replicas). Do try it out and let us know if it helps resolve your issue.

**Reply by mattyoder77** [reply]
Thanks for the link. It doesn't really work for SQL Server as it handles things differently for replicas.
I'll check with the Tiberius repo to see if they can help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
